### DISCUSSION TITLE
Is it possible to use a string variable with drawString()?

### DISCUSSION THREAD

**Discussion by rasmushauschild** [Bodmer/TFT_eSPI#2105] [Q&A - General] [answered]
Hi there!
I'm new to this remarkable library and fairly new to embedded c++ in general, so please bear with me :)
I would like to draw a string (which changes over time) to a sprite, however, I have not been able to get this to work with the `drawString()` function.
When I try I get this error:
`error: no matching function for call to 'TFT_eSprite::drawString(std::__cxx11::string&, int, int&, int)'`

From what I can tell from the error, `drawString()` expects a `const char*`, so my question is how can I go about drawing variable text to a sprite?

Thank you in advance!

**Comment by Bodmer** [comment, marked as answer]
This tutorial should help, it uses the String class which makes building strings out of characters and numbers easy.
https://docs.arduino.cc/built-in-examples/strings/StringConstructors

Based on that tutorial example sketch you could replace the Serial.println(stringOne); for example with:

tft.drawString(stringOne, x, y); // x, y is position on TFT screen

Don't worry about the const qualifier, this simply means the function being called in the library is not allowed to try change the variable.

**Reply by rasmushauschild** [reply]
Amazing! Can't believe I didn't stumble upon that at all - Thank you so much for your thorough answer. I appreciate it! 

Have a great weekend!
